src/node_file.cc

```diff
@@ -93,6 +92,40 @@ class FSReqWrap: public ReqWrap<uv_fs_t> {
 
   size_t self_size() const override { return sizeof(*this); }
 
+  void Resolve(Environment* env, Local<Value> arg) {
+    Local<Context> context = env->context();
+    Local<Value> promise =
+        object()->Get(context, env->oncomplete_string()).ToLocalChecked();
+    if (promise->IsPromise()) {
+      if (promise.As<Promise>()->State() != Promise::kPending) return;
```

Comment:
When is this not true? Can you make this a check or, alternatively, add a comment?

Reply:
+1 .. yeah, was already looking at making this a CHECK. It *should* always be true.